internal/backend/db/dbgorm/vars.go

```diff
@@ -0,0 +1,116 @@
+package dbgorm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"gorm.io/gorm/clause"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/db/dbgorm/scheme"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+func (db *gormdb) SetVars(ctx context.Context, vars []sdktypes.Var) error {
+	if i, err := kittehs.ValidateList(vars, func(_ int, v sdktypes.Var) error {
+		return v.Strict()
+	}); err != nil {
+		return fmt.Errorf("#%d: %w", i, err)
+	}
+
+	return db.transaction(ctx, func(tx *tx) error {
```

Comment:
could we enclose in db.transaction only db-related ops?
e.g. first create (not under transaction) list of scheme.Var with everything precomputed and separated list of indexes + cids to fetch iid(s), 
then under transcation only fetch and update?

Reply:
now getting all connections in a batch